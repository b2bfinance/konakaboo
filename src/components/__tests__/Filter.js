import React from 'react';
import { render, shallow } from 'enzyme';
import {
  Filter,
  FilterWrapper,
  FilterFormControl,
  FilterSelect,
  FilterMenuItem,
  mapDispatchToProps,
  mapStateToProps
} from '../Filter';

test('FilterWrapper renders correctly', () => {
  expect(render(<FilterWrapper />)).toMatchSnapshot();
});

test('FilterFormControl renders correctly', () => {
  expect(shallow(<FilterFormControl />)).toMatchSnapshot();
});

test('FilterSelect renders correctly', () => {
  expect(
    shallow(
      <FilterSelect
        theme={{ filterHeaderBorder: stubData.theme.filterHeaderBorder }}
      />
    )
  ).toMatchSnapshot();
});

test('FilterMenuItem renders correctly', () => {
  expect(
    shallow(
      <FilterMenuItem
        selected
        theme={{
          filterChosenBackground: stubData.theme.filterChosenBackground,
          filterChosenColor: stubData.theme.filterChosenColor
        }}
      />
    )
  ).toMatchSnapshot();
});

test('Filter renders correctly', () => {
  const { chosen, available } = stubData.filters.withSingleAndMultiChoiceChosen;

  expect(
    shallow(<Filter chosen={chosen} filters={available} />)
  ).toMatchSnapshot();
});

test('mapDispatchToProps maps the dispatch correctly', () => {
  expect(Object.keys(mapDispatchToProps)).toEqual(['handleSelectChange']);
});

test('mapStateToProps maps the state correctly', () => {
  const stubState = stubData.filters.withSingleAndMultiChoiceChosen;

  expect(mapStateToProps({ filters: stubState })).toEqual({
    chosen: stubState.chosen,
    filters: stubState.available
  });
});
