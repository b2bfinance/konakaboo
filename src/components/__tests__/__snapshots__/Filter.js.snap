// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
Array [
  <styled.div
    desktop="20"
    key="TEST_KEY_1"
    phone="100"
    tablet="33.333"
/>,
  <styled.div
    desktop="20"
    key="TEST_KEY_2"
    phone="100"
    tablet="33.333"
/>,
]
`;

exports[`FilterFormControl renders correctly 1`] = `
.c0 {
  width: 100%;
}

<withStyles(FormControl)
  className="c0"
/>
`;

exports[`FilterMenuItem renders correctly 1`] = `
.c0 {
  background-color: #009688 !important;
  color: #fff !important;
}

<withStyles(MenuItem)
  className="c0"
  selected={true}
  theme={
    Object {
      "filterChosenBackground": "#009688",
      "filterChosenColor": "#fff",
    }
  }
/>
`;

exports[`FilterSelect renders correctly 1`] = `
.c0:hover:before {
  background-color: #009688 !important;
}

.c0:after {
  background-color: #009688 !important;
}

<withStyles(Select)
  className="c0"
  theme={
    Object {
      "filterHeaderBorder": "#009688",
    }
  }
/>
`;

exports[`FilterWrapper renders correctly 1`] = `
.c0 {
  width: 100%;
  min-height: 1px;
  border-radius: 0.2rem;
  margin-bottom: 1rem;
  position: relative;
}

@media (min-width:36em) {
  .c0 {
    padding-right: 1rem;
  }

  .c0:last-child {
    padding-right: 0;
  }
}

<div
  class="c0"
>
  
</div>
`;
